flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1046,6 +1063,63 @@ class ScalarExprLowering {
       llvm_unreachable("unhandled constant");
     }
   }
+
+  /// Generate a raw literal value and store it in the rawVals vector.
+  template <Fortran::common::TypeCategory TC, int KIND>
+  void
+  genRawLit(const Fortran::evaluate::Scalar<Fortran::evaluate::Type<TC, KIND>>
+                &value) {
+    mlir::Attribute val;
+    if constexpr (TC == Fortran::common::TypeCategory::Integer) {
+      auto rTy = builder.getIntegerType(rawValWidth);
+      val = builder.getIntegerAttr(rTy, value.ToInt64());
+    } else if constexpr (TC == Fortran::common::TypeCategory::Logical) {
+      auto rTy = builder.getIntegerType(rawValWidth);
```

Comment:
I thought of doing it that way, but at some point I thought I ran into an exception.  But perhaps it will just work now that other things are working. I will try it again.